unity/EditorPlugin/AfterUnity56/Navigation/Initialization.cs

```diff
@@ -63,16 +65,17 @@ public static void OnModelInitializationHandler(UnityModelAndLifetime modelAndLi
         }
       });
 
-      modelValue.ShowFileInUnity.Advise(connectionLifetime, result =>
+      modelValue.ShowFileInUnity.AdviseNotNull(connectionLifetime, result =>
       {
-        if (result != null)
+        var matchedUnityPath = AssetDatabase.GetAllAssetPaths()
```

Comment:
This could be a very expensive call, for very large projects. We could get an array of tens of thousands of asset paths. We're also creating 2 `FileInfo` instances for each asset. Could we do this as a fallback if we can't get the path another way?

Reply:
That is definitely possible. I will restart the review again.

Reply:
https://github.com/JetBrains/resharper-unity/pull/2124